forge-gui/res/cardsfolder/upcoming/sanctuary_wall.txt

```diff
@@ -0,0 +1,11 @@
+Name:Sanctuary Wall
+ManaCost:1 W
+Types:Artifact Creature Wall
+PT:0/4
+K:Defender
+A:AB$ Tap | Cost$ 2 W T | ValidTgts$ Creature |  SubAbility$ DBCounter | SpellDescription$ Tap target creature. You may put a stun counter on it. If you do, put a stun counter on CARDNAME. (If a permanent with a stun counter would become untapped, remove one from it instead.)
+SVar:DBCounter:DB$ PutCounter | Defined$ Targeted | CounterType$ Stun | CounterNum$ 1 | Optional$ True | RememberPut$ True | SubAbility$ DBCounterSelf
+SVar:DBCounterSelf:DB$ PutCounter | Defined$ Self | CounterNum$ 1 | CounterType$ Stun | ConditionDefined$ Remembered | ConditionPresent$ GE1 | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True 
+DeckHas:Ability$Counters
+Oracle:Defender\n{2}{W}, {T}: Tap target creature. You may put a stun counter on it. If you do, put a stun counter on Sanctuary Wall. (If a permanent with a stun counter would become untapped, remove one from it instead.)
```

Comment:
@tool4ever is that written as cost?

Reply:
yea I've been a bit lax with that
but this should end up cleaner then

Reply:
The tap ability should be part of the cost ?

Reply:
no, only the first counter

Reply:
something like this ? `AB$ PutCounter | Cost$ AddCounter<1/STUN/Targeted> |`

Reply:
You need to use UnlessCost and also proper valid syntax, try: `Card.targetedBy`

Reply:
_SVar:DBCounter:DB$ PutCounter | UnlessCost$ AddCounter<1/STUN/Card.targetedBy> | UnlessSwitched$ True | Defined$ Self | CounterNum$ 1 | CounterType$ Stun_ 

tried this one without succes

Reply:
maybe you tested it with AI creature since you forgot UnlessPayer$?

Reply:
Thanks that made it work, thanks